include_core/unix/thrdsup.h

```diff
@@ -89,6 +89,10 @@ int linux_pthread_cond_timedwait(pthread_cond_t *cond, pthread_mutex_t *mutex, c
 intptr_t init_thread_library(void);
 intptr_t set_pthread_priority(pthread_t handle, omrthread_prio_t j9ThreadPriority);
 intptr_t set_pthread_name(pthread_t self, pthread_t thread, const char *name);
+intptr_t sem_init_zos(j9sem_t s, int pShared, int initValue);
+intptr_t sem_destroy_zos(j9sem_t s);
+intptr_t sem_wait_zos(j9sem_t s);
+intptr_t sem_post_zos(j9sem_t s);
```

Comment:
can the duplicate prototypes in `thrdsup.h` be removed?
https://github.com/eclipse/omr/blob/4f257e0c2853bf13a4abcd7731ac21dfa9a3bad9/thread/unix/thrdsup.c#L69-L74

Reply:
I had to add these function declarations due to errors like this with Open XL:
```
/jit/team/gauravc/repos/omr/thread/common/j9sem.c:63:8: error: call to undeclared function 'sem_init_zos'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
   63 |                 rc = SEM_INIT(s, 0, initValue);
```

Dependent files like `j9sem.c` were using `thrdsup.h` header includes, and Open XL needs explicit method declarations. Removing this would cause errors here again, would it make sense to remove the prototypes in the `thrdsup.c` file, i.e. would they not be needed anymore since they are already explicitly declared in the header file as per this PR ?

Reply:
yes, the `*.c` definitions will be redundant and not needed since they are declared in the `*.h*' file.

Reply:
Removed the redundant declarations in the  `thrdsup.c` file.